src/mjolnir/landmark_builder.cc

```diff
@@ -168,5 +221,44 @@ std::vector<Landmark> LandmarkDatabase::get_landmarks_in_bounding_box(const doub
   return landmarks;
 }
 
+bool BuildLandmarkFromPBF(const boost::property_tree::ptree& pt,
+                          const std::vector<std::string>& input_files) {
+  // parse config to get landmark database
+  auto db = pt.get_optional<std::string>("landmarks");
+  if (!db) {
+    LOG_ERROR("Landmarks config info not found. Landmarks builder will not be created.");
+    return false;
+  }
+
+  const filesystem::path parent_dir = filesystem::path(*db).parent_path();
+  if (!filesystem::exists(parent_dir) && !filesystem::create_directories(parent_dir)) {
+    LOG_ERROR("Can't create parent directory " + parent_dir.string());
+    return false;
+  }
```

Comment:
ok so we have the removal of existing db in the constructor then we need this create the directory if it doesnt exist also in the constructor, it doesnt make sense for the two things to be decoupled

Reply:
make sense. done.